lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -336,7 +344,7 @@ export class AuthorizationCodeClient extends BaseClient {
         }
 
         // Add scope array, parameter builder will add default scopes and dedupe
-        parameterBuilder.addScopes(request.scopes);
+        parameterBuilder.addScopes(request.scopes, undefined, this.config.authOptions.authority.options.OIDCOptions?.defaultScopes);
```

Comment:
As opposed to my previous comment, here the config value is used in multiple places and there's validation done on whether OIDCOptions is set. This means it's more practical to set `const oidcDefaultScopes = this.config.authOptions.authority.options.OIDCOptions?.defaultScopes` somewhere above this and just reference the value here and in line 507.

Reply:
This way it's more readable and the logical check for OIDCOptions is only executed once.